### ISSUE TITLE
binrefinement checkm error

### ISSUE THREAD

**Issue by dangchenyuan** [bxlab/metaWRAP#144]
I have 3 reassembled bins folds (generated by metawrap ), then I want to use metawrap BIN_REFINEMENT to merge the 3 bins folds, but when I ran the pipeline, these were somethings wrong with the checkm:

*******************************************************************************
Running CheckM on binsAB bins                         
*******************************************************************************
 [CheckM - tree] Placing bins in reference genome tree.
*******************************************************************************
  [Error] No bins found. Check the extension (-x) used to identify bins.

  Controlled exit resulting from an unrecoverable error or warning.

******************************************************************************************************
                       Something went wrong with running CheckM. Exiting...                        

                                            
it seems that the there are no bins readed in binsAB, binsBC, binsAC, binsABC, which resulted in the error.               
All done!
Specified 3 input bin sets: -1 binsA -2 binsB -3 binsC
Add folder/bin name to contig name for binsA bins
Add folder/bin name to contig name for binsB bins
Add folder/bin name to contig name for binsC bins
Combine all bins together
The number of refined bins: 0
Exporting refined bins...

Deleting temporary files

All done!

                there are 0 refined bins in binsAB                                     
                there are 0 refined bins in binsBC                             
                there are 0 refined bins in binsAC                         
               there are 0 refined bins in binsABC                               
----------------------------------------------------------------------------------------------------------
                           Bin refinement finished successfully!                                
----------------------------------------------------------------------------------------------------------

                     fixing bin naming to .fa convention for consistancy...                         
----------------------------------------------------------------------------------------------------------

**Comment by ursky** [comment]
Check that binsA/binsB/binsC output directories have bins in them. Did you use the .fa or .fasta extensions for your fasta files? And having an unedited and **full** output of the program would be helpful.

**Comment by dangchenyuan** [comment]
@ursky yes, binsA, binsB, binsC output directories have bins, but the combined bins (binsAB, binsAC, etc.)output don't have bins.

**Comment by dangchenyuan** [comment]
and the files in the bins output directories are the .fa extensions. I don't know why combined bins output have not results?

**Comment by ursky** [comment]
Is it possible that the 3 bin sets have different contig naming conventions? Its also entirely posssible that the binning_refiner cant work on so few bins. In your case you might be better off just running CheckM and manually choosing the bins with the best stats.

**Comment by ursky** [comment]
I think there is a misunderstanding here. What I meant is if you have more than 3 bin **sets**, then you need to to grouped runs. In you case, you just have 9 samples. It looks like your assembly quality from partial assembly (just 3 samples at a time) is poor due to low coverage, which is why you got so few bins. 

Re-run the assembly with reads concatenated from ALL of the samples at the same time (see example tutorial - this can be with as many samples as you want) to get a co-assembly file. Then use all of the individual samples to help bin this co-assembly with the binning module, to get your bin sets  - metabat/maxbin/concoct - whatever you want. Then re-do a single bin_refinement run.

**Comment by dangchenyuan** [comment]
@ursky thank you, I'll try.

**Comment by ganiatgithub** [comment]
> Is it possible that the 3 bin sets have different contig naming conventions? Its also entirely posssible that the binning_refiner cant work on so few bins. In your case you might be better off just running CheckM and manually choosing the bins with the best stats.

Hi there,
So I have the same issue, I have three bin sets, as A, B, and C, but refined bins in binsABC is empty and the metawrap bin_refinement stops here. 
So if I manually pick good quality bins from A, B and C using checkm, how do I continue with the subsequent consolidate and dereplicate steps?

And any ideas why refined bins in bins BC, AC, and ABC are empty?

Cheers

**Comment by ursky** [comment]
Why are they different? Did the sets originate from different assemblies? If so, bin refinement algorithms will not work. Instead, you need to de-replicate the bins with something like DRep. I need to know a bit more about the problem to give specific advice, however.
